### ISSUE TITLE
NETCore 3.0  AspectInjector : error AI_FAIL

### ISSUE THREAD

**Issue by beckjin** [pamidur/aspect-injector#118]
It was not possible to find any compatible framework version
  The specified framework 'Microsoft.NETCore.App', version '2.1.0' was not found.
    - The following frameworks were found:
        3.0.0 at [/usr/share/dotnet/3.0.100/shared/Microsoft.NETCore.App]
  
  You can resolve the problem by installing the specified framework and/or SDK.
  
  The .NET Core frameworks can be found at:
    - https://aka.ms/dotnet-download
AspectInjector : error AI_FAIL: Aspect Injector processing has failed. See other errors

**Comment by beckjin** [comment]
Linux

**Comment by jeremyj01** [comment]
First of all, this Aspect-Injector is freaking brilliant!  Great job!

I'm having the same challenge when using a multi-stage build in docker based on Alpine Linux.  

I'm trying to attack the problem by creating a docker image that has both SDK's in it, but so far I'm not having any luck.

**Comment by pamidur** [comment]
Thanks for your interest. I'm actually thinking about releasing is standalone without any framework requirements.

Oleksandr Hulyi

________________________________

**Comment by jeremyj01** [comment]
Actually, i was able to create a Dockerfile I could perform a multi-stage build with.  I've created a gist for it here:

https://gist.github.com/jeremyj01/b9b60a96a7498c8625b1192b3a979c08

**Comment by Visagalis** [comment]
Using official MS docker image: **cr.microsoft.com/dotnet/core/sdk:3.1**

can't run CI via Docker:
```
It was not possible to find any compatible framework version
  The framework 'Microsoft.NETCore.App', version '2.1.0' was not found.
    - The following frameworks were found:
        3.1.1 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  
  You can resolve the problem by installing the specified framework and/or SDK.
  
  The specified framework can be found at:
    - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=2.1.0&arch=x64&rid=debian.10-x64
AspectInjector : error AI_FAIL: Aspect Injector processing has failed. See other errors. [/builds/Business/Business.csproj]

Build FAILED.

AspectInjector : error AI_FAIL: Aspect Injector processing has failed. See other errors. [/builds/Business/Business.csproj]
    0 Warning(s)
    1 Error(s)
```

It's quite big issue that this package has a hidden dependency on 'Microsoft.NETCore.App', version '2.1.0'.

**Comment by pamidur** [comment]
Hi,
The AspectInjector is a netcoreapp2.1 executable, thus it requires dotnet core 2.1 installed.

I see that now since netcore 3.0 is release it starting to be an issue for netcore3.0 only containers. So I'll be looking on how to resolve it. 
Until then you might alter you dockerfile in a way it installs netcore2.1 before compiling your app

**Comment by pamidur** [comment]
@Visagalis , just to make it clear, it is not a hidden dependency. At least not at runtime. Netcore2.1 is required only to build time. This dependency doesn't leak to your code after it is built.

**Comment by twistingmercury** [comment]
@Visagalis I ran into this issue a few months back.  I created a gist that will resolve the issue int a Dockerfile:

https://gist.github.com/jeremyj01/b9b60a96a7498c8625b1192b3a979c08

**Comment by pamidur** [comment]
@Visagalis , @twistingmercury , could you please try version 2.2.6. I made it explicitly allowed to use any major runtime version installed. Let's hope new netcore runtimes don't bring breaking changes)

**Comment by jeremyj01** [comment]
Sure thing.  I’ll get to it this afternoon (PST).

-JJ

**Comment by twistingmercury** [comment]
@pamidur it's working for me against .netcore 3.1!  Thanks!

**Comment by pamidur** [comment]
fixed in 2.2.6
